packages/flutter/lib/src/rendering/error.dart

```diff
@@ -50,6 +51,9 @@ class RenderErrorBox extends RenderBox {
 
   /// The message to attempt to display at paint time.
   final String message;
+  final double systemTopPadding = Zone.current['systemTopPadding'] == null
```

Comment:
Processing the info a bit :)
My 2 cents:
- In the very long term, I mostly agree with Ian's aesthetic views. Avoiding global statics will likely save us a bunch of trouble down the road. Along the axis of strict disciplined API <-> Python-like magical experience, using something like https://github.com/angular/di.dart or letting RenderObjects (maybe?) retrieve the Window object from the this. instance rather than from a global is probably more predictable. Everyone reading the getter and the setter knows it's overridable. Zone getter overrides is probably in the middle since it seems like a flat dictionary of overrides. Import overrides is probably more magical. Works in this case but if promoted as a generic pattern, developers could end up with a number of surprises. 

- For this very particular case, we may be over engineering this. If we add smarts to window padding, we might not want to run it for every 16ms, especially considering the dependency reduction we might want to have on this file. If it's not smart, we might as well shed the dependency entirely and hard-code an offset.